Tools/autotest/common.py

```diff
@@ -979,15 +979,18 @@ def detect_and_handle_reboot(self, old_bootcount, required_bootcount=None, timeo
 
         self.initialise_after_reboot_sitl()
 
-    def set_streamrate(self, streamrate):
+    def set_streamrate(self, streamrate, timeout=10):
         tstart = time.time()
         while True:
-            if time.time() - tstart > 10:
+            if time.time() - tstart > timeout:
                 raise AutoTestTimeoutException("stream rate change failed")
 
             self.mavproxy.send("set streamrate %u\n" % (streamrate))
             self.mavproxy.send("set streamrate\n")
-            self.mavproxy.expect('.*streamrate ((?:-)?[0-9]+)', timeout=1)
```

Comment:
why do we need mavproxy for setting the streamrate ? can't we have a direct method to doing it or I miss something ?

Reply:
I do have such an implementation as part of my autotest-direct-sitl branch.

Reply:
maybe I would be worth it to do direct sitl connexion for something like that !